src/riak_kv_index_hashtree.erl

```diff
@@ -619,6 +659,14 @@ do_build_finished(State=#state{index=Index, built=_Pid}) ->
     riak_kv_entropy_info:tree_built(Index, BuildTime),
     State#state{built=true, build_time=BuildTime, expired=false}.
 
+do_inform_waiting(#state{waiting_for_lock = {Type, Waiting}} = State) when is_pid(Waiting) ->
+    {ok, State2} = do_get_lock(Type, Waiting, State#state{waiting_for_lock = undefined}),
+    Waiting ! {hashtree_lock, State2#state.index},
+    State;
```

Comment:
We are returning State without removing the waiting_for_lock field (that was done to State2).